### DISCUSSION TITLE
webp support in 6.7.7-10

### DISCUSSION THREAD

**Discussion by ychaouche** [ImageMagick/ImageMagick6#298] [General] [answered]
### ImageMagick version

6.7.7-10 

### Operating system, version and so on

Linux Mint 17 Qiana

### Description

Tried to convert webp to png and failed.  
Installed the webp package as well as the libwebp-dev library  
(just in case)  
I copied the libweb.so lib to the modules.../coders/ directory.  

`$ sudo cp /usr/lib/x86_64-linux-gnu/libwebp.so /usr/lib/x86_64-linux-gnu/ImageMagick-6.7.7/modules-Q16/coders/webp.so`

I also found a webp.la file on [github](https://github.com/archfz/word-template/blob/b2c1f8d78ad68f6b4c46c51b82c1aba8321383bd/native/lib/coders/webp.la),
which I copied to  /usr/lib/x86_64-linux-gnu/ImageMagick-6.7.7/modules-Q16/coders/webp.la  
[and modified](https://gist.github.com/ychaouche/cf15bc42044176a65cba8a4a9905d0f7) to fit my installation paths and version numbers.

It still says no delegate found for this image type

```
$ convert ./IN/severalcenturieslater.webp ./IN/severalcenturieslater.png
convert.im6: no decode delegate for this image format `./IN/severalcenturieslater.webp' @ error/constitute.c/ReadImage/544.
convert.im6: no images defined `./IN/severalcenturieslater.png' @ error/convert.c/ConvertImageCommand/3044.
$
$ convert -version
Version: ImageMagick 6.7.7-10 2018-09-28 Q16 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2012 ImageMagick Studio LLC
Features: OpenMP

$ 
```

[Here's an strace of it](https://gist.github.com/ychaouche/cf15bc42044176a65cba8a4a9905d0f7#file-strace-convert)  
(strace -fyvs 72 -e file)

**Comment by dlemstra** [comment]
You should not be using that ancient version anymore and upgrade to a more recent version.

**Reply by ychaouche** [reply]
Thanks @dlemstra  
this is obviously the best thing to do.  
I would be grateful to get a pointer on what part of the documentation (or source code) I must read to understand the error and its implications.

**Reply by dlemstra** [reply, marked as answer]
When you look at the version you will notice that there is no `delegates` path. This is the output on my (Windows) machine:

```
Version: ImageMagick 7.1.1-25 (Beta)  Q16-HDRI x64 d1cd21ccc:20231221 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Debug Channel-masks(64-bit) Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo fftw flif freetype gslib heic jbig jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193833133)
```

Your build did not include any delegates so that means your version can only read the build in formats that don't require external libraries. You will probably need to recompile ImageMagick.

**Reply by ychaouche** [reply]
thanks again @dlemstra.  
Originally, I thought that imagemagick would dynamically load modules from the its `modules-<...>/coders/ `directory,
so that if I add anything there it would load it and use it.  
Apparently, it's not the case.
